src/simulations/collect.jl

```diff
@@ -164,6 +165,121 @@ function run!(
     return df_agent, df_model
 end
 
+"""
+    offline_run!(model, agent_step! [, model_step!], n::Integer; kwargs...) → agent_df, model_df
+    offline_run!(model, agent_step!, model_step!, n::Function; kwargs...) → agent_df, model_df
+
+Run the model while dumping data to CSV files. See [`run!`](@ref) for details.
+
+## Keywords
+* `writing_interval=1` : write to file every `writing_interval` times the data is collected. Small values have larger effects on performance.
```

Comment:
I don't understand what this is from its documentation. How does this place with the `when` to collect keywords of `run!`? Also please keep lines of code within 92 characters max.